app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -30,7 +30,32 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
       firstScript.parentNode.insertBefore(newScript, firstScript)
     },
 
+    ensureIndexesArePopulated: function (data) {
+      if (!data.event_data) {
+        return data
+      }
+
+      if (!data.event_data.index) {
+        return data
+      }
```

Comment:
Any chance this if and the previous if could be combined? I suspect not but wanted to check.

Reply:
It can't as if `event_data` is `undefined`, and you try and call `.index` on it, you will get an error essentially saying `can't call .index on something that's undefined`.